src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/CustomerGroupsController.php

```diff
@@ -0,0 +1,117 @@
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Controller\Admin\Configure\ShopParameters;
+
+use PrestaShop\PrestaShop\Core\Search\Filters\CustomerGroupsFilters;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Controller responsible for "Configure > Shop Parameters > Customer Settings > Groups" page.
+ */
+class CustomerGroupsController extends FrameworkBundleAdminController
+{
+    /**
+     * Show Groups tab.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))", message="Access denied.")
+     *
+     * @param Request $request
+     * @param CustomerGroupsFilters $filters
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, CustomerGroupsFilters $filters): Response
+    {
+        $customerGroupsGridFactory = $this->get('prestashop.core.grid.factory.customer_groups');
+        $customerGroupsGrid = $customerGroupsGridFactory->getGrid($filters);
+
+        return $this->render('@PrestaShop/Admin/Configure/ShopParameters/CustomerSettings/Groups/index.html.twig', [
+            'customerGroupsGrid' => $this->presentGrid($customerGroupsGrid),
+            'layoutTitle' => $this->trans('Groups', 'Admin.Navigation.Menu'),
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'enableSidebar' => true,
+        ]);
+    }
+
+    /**
+     * Displays and handles customer group form.
+     *
+     * @AdminSecurity(
+     *     "is_granted('create', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_customer_groups_index",
+     *     message="You need permission to create this."
+     * )
+     *
+     * @return Response
+     */
+    public function createAction(): Response
+    {
+        return $this->redirect(
+            $this->getContext()->link->getAdminLink(
+                'AdminGroups',
+                true,
+                [],
+                [
+                    'addgroup' => '',
+                ]
+            )
+        );
+    }
+
+    /**
+     * Displays title form.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_customer_groups_index",
+     *     message="You need permission to edit this."
+     * )
+     *
+     * @param int $groupId
+     *
+     * @return Response
+     */
+    public function editAction(int $groupId): Response
+    {
+        return $this->redirect(
+            $this->getContext()->link->getAdminLink(
+                'AdminGroups',
+                true,
+                [],
+                [
+                    'updategroup' => '',
+                    'id_group' => $groupId,
+                ]
+            )
+        );
```

Comment:
Can you somehow create an adapter compatible adapter for $this, so we can get rid of the context in the controller ?

Reply:
(and for the above function as well 👍 )

Reply:
This is temporary before the CQRS to "wrap" the legacy forms to edit/delete/create before migrating them, it will be removed in following PRs

Reply:
ok

Reply:
![image](https://user-images.githubusercontent.com/28731165/216629646-abe8bbff-4a0d-49cf-8157-328d2cf8daba.png)